components/locale/src/parser/langid.rs

```diff
@@ -0,0 +1,102 @@
+use std::iter::Peekable;
+
+pub use super::errors::ParserError;
+use crate::subtags;
+use crate::LanguageIdentifier;
+
+#[derive(PartialEq)]
+pub enum ParserMode {
+    LanguageIdentifier,
+    Locale,
+    Partial,
+}
+
+pub fn parse_language_identifier_from_iter<'a>(
+    iter: &mut Peekable<impl Iterator<Item = &'a [u8]>>,
+    mode: ParserMode,
+) -> Result<LanguageIdentifier, ParserError> {
+    let language;
+    let mut script = None;
+    let mut region = None;
+    let mut variants = Vec::new();
+
+    let mut position;
+
+    if let Some(subtag) = iter.next() {
+        if let Ok(s) = subtags::Language::from_bytes(subtag) {
+            language = s;
+            position = 1;
```

Comment:
Would using enums (Language, Script, Region vs 1, 2, 3) be better than magic numbers?

Or is position variable being used for something else...